src/handlers/artist.rs

```diff
@@ -0,0 +1,206 @@
+                );
+                artist.selected_related_artist_index = next_index;
+            }
+            ArtistBlock::Empty => {}
+        }
+    }
+}
+
+fn handle_down_press_on_hovered_block(app: &mut App) {
+    if let Some(artist) = &mut app.artist {
+        match artist.artist_hovered_block {
+            ArtistBlock::TopTracks => {
+                artist.artist_hovered_block = ArtistBlock::Albums;
+            }
+            ArtistBlock::Albums => {
+                artist.artist_hovered_block = ArtistBlock::RelatedArtists;
+            }
+            ArtistBlock::RelatedArtists => {
+                artist.artist_hovered_block = ArtistBlock::TopTracks;
+            }
+            ArtistBlock::Empty => {}
+        }
+    }
+}
+
+fn handle_up_press_on_selected_block(app: &mut App) {
+    if let Some(artist) = &mut app.artist {
+        match artist.artist_selected_block {
+            ArtistBlock::TopTracks => {
+                let next_index = common_key_events::on_up_press_handler(
+                    &artist.top_tracks,
+                    Some(artist.selected_top_track_index),
+                );
+                artist.selected_top_track_index = next_index;
+            }
+            ArtistBlock::Albums => {
+                let next_index = common_key_events::on_up_press_handler(
+                    &artist.albums.items,
+                    Some(artist.selected_album_index),
+                );
+                artist.selected_album_index = next_index;
+            }
+            ArtistBlock::RelatedArtists => {
+                let next_index = common_key_events::on_up_press_handler(
+                    &artist.related_artists,
+                    Some(artist.selected_related_artist_index),
+                );
+                artist.selected_related_artist_index = next_index;
+            }
+            ArtistBlock::Empty => {}
+        }
+    }
+}
+
+fn handle_up_press_on_hovered_block(app: &mut App) {
+    if let Some(artist) = &mut app.artist {
+        match artist.artist_hovered_block {
+            ArtistBlock::TopTracks => {
+                artist.artist_hovered_block = ArtistBlock::RelatedArtists;
+            }
+            ArtistBlock::Albums => {
+                artist.artist_hovered_block = ArtistBlock::TopTracks;
+            }
+            ArtistBlock::RelatedArtists => {
+                artist.artist_hovered_block = ArtistBlock::Albums;
+            }
+            ArtistBlock::Empty => {}
+        }
+    }
+}
+
+fn handle_enter_event_on_selected_block(app: &mut App) {
+    if let Some(artist) = &mut app.artist.clone() {
+        match artist.artist_selected_block {
+            ArtistBlock::TopTracks => {
+                let selected_index = artist.selected_top_track_index;
+                let selected_track = &artist.top_tracks[selected_index];
+                app.start_playback(None, Some(vec![selected_track.uri.to_owned()]), Some(0));
```

Comment:
We could pass in an array of `uri`s with this selected track at the head? This would allow spotify to continue playing through the top tracks.

Reply:
Thank you for code review and give me constructive comments.
I modified relevant part as following:
```
                let top_tracks = artist
                    .top_tracks
                    .iter()
                    .map(|track| track.uri.to_owned())
                    .collect();
                app.start_playback(None, Some(top_tracks), Some(selected_index));
```